test/e2e/e2e_upgrade_test.go

```diff
@@ -128,6 +146,26 @@ var _ = Describe("Workload cluster creation", func() {
 				ControlPlane: client.ObjectKeyFromObject(result.ControlPlane),
 			}, e2eConfig.GetIntervals(specName, "wait-control-plane")...)
 
+			WaitForClusterReady(ctx, WaitForClusterReadyInput{
+				Getter:    bootstrapClusterProxy.GetClient(),
+				Name:      result.Cluster.Name,
+				Namespace: result.Cluster.Namespace,
+			}, e2eConfig.GetIntervals(specName, "wait-cluster")...)
+
+			By("Verifying machine rollout did not happen")
+			updatedMachineList := GetMachinesByCluster(ctx, GetMachinesByClusterInput{
+				Lister:      bootstrapClusterProxy.GetClient(),
+				ClusterName: result.Cluster.Name,
+				Namespace:   result.Cluster.Namespace,
+			})
+			Expect(updatedMachineList.Items).ShouldNot(BeEmpty(), "There must be at least one Machine after provider upgrade")
+			updatedMachinesNames := []string{}
+			for _, machine := range updatedMachineList.Items {
+				updatedMachinesNames = append(updatedMachinesNames, machine.Name)
+			}
+			Expect(updatedMachinesNames).Should(ContainElements(machinesNames), "Machines should not have been rolled out after provider upgrade")
+			Expect(len(updatedMachinesNames)).Should(Equal(len(machinesNames)), "Number of Machines should match after provider upgrade")
+
```

Comment:
Can you please wrap this into consistently block to test that it stays unchanged for some time like 2 minutes?